streams/src/main/java/org/apache/kafka/streams/state/internals/StoreQueryUtils.java

```diff
@@ -146,7 +146,7 @@ public static <R> QueryResult<R> handleBasicQueries(
                 "Handled in " + store.getClass() + " in " + (System.nanoTime() - start) + "ns"
             );
         }
-        result.setPosition(position);
+        result.setPosition(position.copy());
```

Comment:
Nice find! But I'm not sure, whether this thread safe now? It seems I'd have to put a lock somewhere to make sure `position` isn't modified before I reach this point.

Reply:
Better yet, we could make the copy at the beginning of this method or even on the caller side.

Reply:
Thanks. I was hoping it would be thready-safe already -- maybe not. Let me double check the code and figure it out.